### DISCUSSION TITLE
The ability to apply global class(es) with cli

### DISCUSSION THREAD

**Discussion by ColaIan** [marp-team/marp#404] [Ideas] [answered]
I'm currently setting up a script to convert marp into pdf files, I'm thinking if there is a way to for example run a command to compile light mode pdfs, then run another command to apply a class (e.g. "dark") to compile the same marp files into dark mode pdfs without editing each and every marp file or creating another dark mode theme file of the same name.

**Comment by yhatt** [comment, marked as answer]
One of idea is using Marpit (markdown-it) plugin via engine customization.

```javascript
// engine.js
const marpGlobalClassPlugin = (md) => {
  md.core.ruler.after(
    'marpit_directives_apply',
    'marp_global_class',
    (state) => {
      if (state.inlineMode) return

      const globalClass = process.env.MARP_GLOBAL_CLASS
      if (!globalClass) return

      for (const token of state.tokens) {
        if (token.type === 'marpit_slide_open') {
          token.attrJoin('class', globalClass)
        }
      }
    }
  )
}

module.exports = ({ marp }) => marp.use(marpGlobalClassPlugin)
```

```bash
# No global class
marp --engine engine.js --pdf markdown.md

# Add global class "dark" to every pages
MARP_GLOBAL_CLASS=dark marp --engine engine.js --pdf markdown.md
```

**Reply by ColaIan** [reply]
Thanks a lot for the help! Just had a try and it's working perfectly, wondering if it's worth adding to marp-cli as a built-in feature lol
